airflow/api_connexion/endpoints/role_and_permission_endpoint.py

```diff
@@ -53,6 +55,39 @@ def _check_action_and_resource(sm: AirflowSecurityManager, perms: list[tuple[str
             raise BadRequest(detail=f"The specified resource: {resource!r} was not found")
 
 
+def _check_permissions_to_revoke(
+    *,
+    sm: AirflowSecurityManager,
+    role_permissions: set[tuple[str, str]],
+    permission_pairs_to_revoke: list[tuple[str, str]],
+    role_name: str,
+) -> None:
+    """
+    Firstly, checks if all the actions and resources exist and otherwise raise 400.
+    Secondly, checks if all the permissions (i.e., action + resource pairs) are assigned to the role,
+    otherwise raises 409.
```

Comment:
Personally my preference is between 404 and 204 actually; Conflict doesn’t seem like the right word for this and instead sound like the revocation failed due to the permission being needed by another resource.

Reply:
I think if we look at a permission as the target resource here, then your reasoning makes perfect sense to me. I was rather looking at this more as changing a part of a role's state.

I'm happy to move to `404`, though wondering if you know anybody from the community, who may want to chime in? So that I don't need to migrate back and forth :)